### PR TITLE
add message tracing id for rest in mng sdk

### PR DESCRIPTION
**Pull request by wanlwanl**

### CODE REVIEW

File: src/Microsoft.Azure.SignalR.Management/RestApiProviderExtensions.cs [hunk 1]
```diff
@@ -28,5 +30,20 @@ public static async Task OrTimeout(this Task task, CancellationToken cancellatio
                 }
             }
         }
+
```
Comment (Y-Sindo) [inline src/Microsoft.Azure.SignalR.Management/RestApiProviderExtensions.cs:33]: Why rename the file? `OrTimeout` is used in `NegotiateProcessor` and `ServiceHubContextBuilder`. It seems nothing to do with `WithTracingIdAsync`

Comment (wanlwanl) [reply src/Microsoft.Azure.SignalR.Management/RestApiProviderExtensions.cs:33]: by accident...

File: src/Microsoft.Azure.SignalR.Management/TaskExtensions.cs [hunk 2]
```diff
@@ -28,5 +30,20 @@ public static async Task OrTimeout(this Task task, CancellationToken cancellatio
                 }
             }
         }
+
+        public static async Task<RestApiEndpoint> WithTracingIdAsync(this Task<RestApiEndpoint> task, bool enable = false)
```
Comment (Y-Sindo) [inline src/Microsoft.Azure.SignalR.Management/TaskExtensions.cs:34]: Given the file name is `TaskExtensions`, this method does not belong to here?

Comment (Y-Sindo) [reply src/Microsoft.Azure.SignalR.Management/TaskExtensions.cs:34]: I think you can move the method to `RestClient.SendAsync`, just pass one more flag.

Comment (vicancy) [inline src/Microsoft.Azure.SignalR.Management/TaskExtensions.cs:34]: why we need `enable` parameter? isn't `WithTracingIdAsync` meaning enable?

Comment (wanlwanl) [reply src/Microsoft.Azure.SignalR.Management/TaskExtensions.cs:34]: going to put the `enable` to `RestClient.SendAsync`

File: src/Microsoft.Azure.SignalR.Common/Utilities/RestClient.cs [hunk 3]
```diff
@@ -152,5 +159,15 @@ private HttpRequestMessage GenerateHttpRequest(string url, IDictionary<string, S
             request.Content = new StringContent(JsonConvert.SerializeObject(payload, _jsonSerializerSettings), Encoding.UTF8, "application/json");
             return request;
         }
+
+        public void AddTracingId(RestApiEndpoint api)
```
Comment (Y-Sindo) [inline src/Microsoft.Azure.SignalR.Common/Utilities/RestClient.cs:163]: No need to public？

File: src/Microsoft.Azure.SignalR.Management/RestHubLifetimeManager.cs [hunk 4]
```diff
@@ -23,14 +23,16 @@ internal class RestHubLifetimeManager : HubLifetimeManager<Hub>, IServiceHubLife
         private readonly string _productInfo;
         private readonly string _hubName;
         private readonly string _appName;
+        private readonly bool _enableMessageTracing;
 
-        public RestHubLifetimeManager(string hubName, ServiceEndpoint endpoint, string productInfo, string appName, RestClient restClient)
+        public RestHubLifetimeManager(string hubName, ServiceEndpoint endpoint, string productInfo, string appName, RestClient restClient, bool enableMessageTracing)
```
Comment (Y-Sindo) [inline src/Microsoft.Azure.SignalR.Management/RestHubLifetimeManager.cs:28]: remove?

File: src/Microsoft.Azure.SignalR.Common/Utilities/RestClient.cs [hunk 5]
```diff
@@ -19,17 +19,20 @@ internal class RestClient
     {
         private readonly JsonSerializerSettings _jsonSerializerSettings;
         private readonly IHttpClientFactory _httpClientFactory;
+        private readonly bool _enableMessageTracing;
 
-        public RestClient(IHttpClientFactory httpClientFactory, JsonSerializerSettings jsonSerializerSettings)
+        public RestClient(IHttpClientFactory httpClientFactory, JsonSerializerSettings jsonSerializerSettings, bool enableMessageTracing = false)
```
Comment (Y-Sindo) [inline src/Microsoft.Azure.SignalR.Common/Utilities/RestClient.cs:24]: No need to make it optional?

**Review by vicancy — APPROVED** [review]

**Review by vicancy — APPROVED** [review]

**Pull request merged**